esmvalcore/local.py

```diff
@@ -136,22 +137,21 @@ def _get_start_end_date(
         and Path(file).exists()
     ):
         logger.debug("Must load file %s for daterange ", file)
-        cubes = iris.load(file)
-
-        for cube in cubes:
-            logger.debug(cube)
-            try:
-                time = cube.coord("time")
-            except iris.exceptions.CoordinateNotFoundError:
-                continue
+        dataset = Dataset(file)
+        if (
+            "time" in dataset.variables
```

Comment:
This may not be as comprehensive as `iris.cube.Cube.coord("time")`. Looking through the Iris code a bit, I suspect that function checks also if one of several variable attributes matches "time": https://github.com/SciTools/iris/blob/1285efbea1d852d6d902abd8a9a1cc55272684c8/lib/iris/common/metadata.py#L1191-L1211

Reply:
Good point, will change that!